src/core/include/openvino/core/validation_util.hpp

```diff
@@ -196,6 +196,56 @@ OPENVINO_API bool are_unique(const std::vector<int64_t>& data);
 /// \param max    Maximum value boiund
 ///
 /// \return Value if between min, max otherwise min or max.
-OPENVINO_API
-int64_t clip(const int64_t& value, const int64_t& min, const int64_t& max);
+OPENVINO_API int64_t clip(const int64_t& value, const int64_t& min, const int64_t& max);
+
+OPENVINO_API bool could_propagate(const Output<Node>& output, std::vector<Node*>& order);
+
+/// \brief Checks if all the elements of the bound Tensor are positive
+OPENVINO_API bool tensor_is_positive(const Tensor& bound);
+
+/// \brief Estimates upper bound for node output tensors using only upper bounds of the nodes
+/// inputs.
+/// \param node Operation to be performed
+/// \param output_values Vector of Tensors representing resulting upper value estimations
+/// \return boolean status if value evaluation was successful.
+OPENVINO_API bool default_upper_bound_evaluator(const Node* node, TensorVector& output_values);
+
+/// \brief Estimates lower bound for node output tensors using only lower bounds of the nodes
+/// inputs.
+/// \param node Operation to be performed
+/// \param output_values Vector of Tensors representing resulting lower value estimations
+/// \return boolean status if value evaluation was successful.
+OPENVINO_API bool default_lower_bound_evaluator(const Node* node, TensorVector& output_values);
+
+/// \brief Evaluates lower value estimation of the output tensor. Traverses graph up to deduce
+/// estimation through it.
+/// \param Node output pointing to the tensor for estimation.
+/// \return Tensor to estimated value.
+OPENVINO_API Tensor evaluate_lower_bound(const Output<Node>& output);
+
+/// \brief Evaluates lower value estimation of the output tensor. Traverses graph up to deduce
+/// estimation through it.
+/// \param output Tensor to be estimated.
+/// \return Tensor to estimated value.
```

Comment:
Why do we need to have this API in public? 
Can it be a part of developer API?

Reply:
I'm not sure about but it looks like could be dev API part. When I modified these functions to use ov::Tensor I decide to move them from ngraph to ov namespace and move between `validation_util.hpp` files.
I've done some check if possible but when there function (or whole file) is in dev API it cannot be included in the version for ngraph namespace and all usage in all cpp file has to be updated. It looks like a lot of changes which is not main goal of this PR so would like to postpone these changes.
I will add task to shape inference improvements to move these functions to dev API and do it in separate PR.

Reply:
So, in general i don't want to introduce these new functions to the public API. Because in this case we cannot remove it without deprecation (I mean released API) and we can forget to move all this API to dev before the release

Reply:
Ok, I moved these new added functions to dev API.